### PR TITLE
add broker middleware to secure endpoints and GUI

### PR DESCRIPTION
**Pull request by adewunmigenius**

### CODE REVIEW

**Review by Rotorsoft — COMMENTED** [review]
Nice job! Just a couple of things so far:
- In subscriptions-form you should disable "submit"
- In all forms you should disable "edit" property when not admin
- What do you think of having a util function isAdmin() to encapsulate the rules in case those change?

**Review by Rotorsoft — COMMENTED** [review]
disable add button in services.hbs when not admin

File: libs/eventually/src/utils.ts [hunk 1]
```diff
@@ -212,3 +213,14 @@ export const validateMessage = (
...
+/**
+ * Validates admin user
+ *
+ * @param req Request payload
+ * @returns boolean when is an Admin
+ */
+export const isAdmin = (req: Request): boolean | undefined => {
+  const { user } = req as Request & { user: Actor };
+  return user && user.roles.includes("admin");
+};
```
Comment (Rotorsoft) [inline libs/eventually/src/utils.ts:226]: this shouldn't be here... eventually doesn't know about express or any other concrete implementation.
eventually-broker on the other hand can provide its own utils and the concept of Admin is only a concern of this package

File: libs/eventually-broker/src/routes/subscriptions.ts [hunk 2]
```diff
@@ -159,10 +159,10 @@ router.get("/:id", async (req, res) => {
           ...props,
           ...sub
         })
-      : res.render("edit-subscription", { ...props, ...err });
+      : res.render("edit-subscription", { ...props, ...err, isAdmin: isAdmin(req)});
   } catch (error) {
     log().error(error);
-    res.render("edit-subscription", { ...props, ...err });
+    res.render("edit-subscription", { ...props, ...err, isAdmin });
```
Comment (Rotorsoft) [inline libs/eventually-broker/src/routes/subscriptions.ts:165]: property or function? why not declare a const isAdmin = utils.isAdmin(req) and reuse it?

Comment (adewunmigenius) [reply libs/eventually-broker/src/routes/subscriptions.ts:165]: I forgot to update this line, will update it now

File: libs/eventually-broker/src/routes/services.ts [hunk 3]
```diff
@@ -60,11 +63,11 @@ router.get("/:id", async (req, res) => {
   try {
     const [service] = await subscriptions().loadServices(id);
     service
-      ? res.render("edit-service", { ...service })
-      : res.render("edit-service", { ...err });
+      ? res.render("edit-service", { ...service, isAdmin: isAdmin(req) })
+      : res.render("edit-service", { ...err, iisAdmin: isAdmin(req) });
```
Comment (Rotorsoft) [inline libs/eventually-broker/src/routes/services.ts:67]: IIS? internet information service from microsoft? 😆

File: libs/eventually-broker/src/utils.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * Validates admin user
+ *
+ * @param req Request payload
+ * @returns boolean when is an Admin
+ */
+export const isAdmin = (req: Request): boolean | undefined => {
+  const { user } = req as Request & { user: Actor };
+  return user && user.roles.includes("admin");
```
Comment (Rotorsoft) [inline libs/eventually-broker/src/utils.ts:12]: user?.roles? to avoid runtime errors?

**Review by Rotorsoft — APPROVED** [review]
LGTM 🚀

**Pull request merged**